src/hotspot/share/opto/loopopts.cpp

```diff
@@ -3969,7 +3969,7 @@ bool PhaseIdealLoop::duplicate_loop_backedge(IdealLoopTree *loop, Node_List &old
     exit_test = back_control->in(0)->as_If();
   }
 
-  if (idom(region)->is_Catch()) {
+  if (idom(region)->is_Catch() || region == head) {
```

Comment:
That check could be done earlier I think at the test for:
 `if (!incr->is_Phi()) {`

Reply:
Thank your for the suggestion! I have move the check to `if (!incr->is_Phi())`